MAVProxy/modules/mavproxy_adsb.py

```diff
@@ -186,18 +204,17 @@ def mavlink_packet(self, m):
                     from MAVProxy.modules.lib import mp_menu
                     from MAVProxy.modules.mavproxy_map import mp_slipmap
                     self.threat_vehicles[id].menu_item = mp_menu.MPMenuItem(name=id, returnkey=None)
-                    if m.emitter_type >= 100 and m.emitter_type-100 in obc_icons:
-                        icon = mp.map.icon(obc_icons[m.emitter_type-100])
-                        threat_radius = get_threat_radius(m.emitter_type-100)
-                    else:
-                        icon = mp.map.icon(self.threat_vehicles[id].icon)
-                        threat_radius = 0
+
+                    threat_radius = get_threat_radius(m.emitter_type, m.squawk)
+                    selected_icon = get_threat_icon(m.emitter_type, id)
                     popup = mp_menu.MPMenuSubMenu('ADSB', items=[self.threat_vehicles[id].menu_item])
-                    # draw the vehicle on the map
-                    mp.map.add_object(mp_slipmap.SlipIcon(id, (m.lat * 1e-7, m.lon * 1e-7),
-                                                                    icon, layer=3, rotation=m.heading*0.01, follow=False,
-                                                                    trail=mp_slipmap.SlipTrail(colour=(0, 255, 255)),
-                                                                    popup_menu=popup))
+                    
+                    if selected_icon is not None:
+                        # draw the vehicle on the map
+                        mp.map.add_object(mp_slipmap.SlipIcon(id, (m.lat * 1e-7, m.lon * 1e-7),
+                                                                        mp.map.icon(selected_icon), layer=3, rotation=m.heading*0.01, follow=False,
```

Comment:
formatting fix required

Reply:
it matches previous, but sure I'll fix that for you.